pkg/cmd/server/origin/auth.go

```diff
@@ -313,9 +313,12 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 				return nil, err
 			}
 
+			// Default state builder, combining CSRF and return URL handling
```

Comment:
the state is intended to be opaque, so only the state object (or an error handler built in concert with the state object) should know that something in the state might be a URL to redirect to.